### DISCUSSION TITLE
how to select the first row and remove some lines

### DISCUSSION THREAD

**Discussion by chanpreet90** [jsvine/pdfplumber#783] [Q&A]
I have a PDF with a table that extends numerous pages, but the first row of every page (except the first page) is not coming. 
I discovered that the first row is not being picked after testing it out using visual debugging. 

1) How can I choose the top row of each page?
2) As you can see in the image below, I drew an arrow pointing to the cells which has extra partition line. 
    How to get rid of that as well.

![download](https://user-images.githubusercontent.com/86623814/209079265-756d0fab-1f85-4aca-8511-952e65bb11aa.png)

**Comment by jsvine** [comment]
Re. 1: The top row isn't recognized as a row because there's no line closing it off at the top. One way to handle this is by programmatically identifying where that line should be and passing it to `explicit_horizontal_lines`. See here for an example: https://github.com/jsvine/pdfplumber/discussions/625

Re. 2: Without access to the PDF itself, it's difficult to know for sure, but my initial suggestion would be to examine the `Page.lines` or `Page.rects` values, see what makes those unwanted lines distinct (e.g., possibly width or color), and then use `Page.filter` to remove them.

**Reply by chanpreet90** [reply]
I have followed post #625, I am still unable to get the required outcome. Would you kindly assist me?

**Reply by samkit-jain** [reply]
That code will not work as-is. You will need to modify it to adjust your needs so that it gives you the top row and not the bottom one. If you have done the modification, please share the PDF and also the code that can be used to reproduce the issue.

**Comment by samkit-jain** [comment]
To get the top horizontal line, you can
1. Crop the page to remove the text at the top of the page and keep only the table and use the `text` strategy.
2. Use a very high value for `join_y_tolerance`.

For removing the extra lines
1. Can try using the `edge_min_length` property.
3. Can try using the `snap_tolerance` property.
4. Can filter out the invisible lines as shared in https://github.com/jsvine/pdfplumber/issues/285#issuecomment-710111180
